engine/pkg/log/util/util.go

```diff
@@ -35,33 +33,29 @@ const (
 	constFieldWorkerKey = "worker_id"
 )
 
-// L is the alias of NewLogger4Framework for easy use
-func L() *zap.Logger {
-	return NewLogger4Framework()
-}
-
-// NewLogger4Framework return a new logger for framework
-func NewLogger4Framework() *zap.Logger {
-	return log.L().With(
-		zap.Bool(constFieldFrameworkKey, constFieldFrameworkValue),
-	)
-}
-
-// NewLogger4Master return a new logger for master
-func NewLogger4Master(project tenant.ProjectInfo, masterID frameModel.MasterID) *zap.Logger {
-	return log.L().With(
+// WithProjectInfo attaches project info to context logger
+func WithProjectInfo(ctx context.Context, project tenant.ProjectInfo) context.Context {
```

Comment:
It looks weird that `logutil.WithProjectInfo` returns a `context.Context`. I think `NewContextWithXXX` would be better.  See go's example: https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/context/context.go;l=144.

Reply:
BTW, why separate these functions into a new package `util`?